tests/foreman/ui/test_computeresource_vmware.py

```diff
@@ -501,3 +502,110 @@ def test_positive_access_vmware_with_custom_profile(session, module_vmware_setti
+    ptable = target_sat.api.PartitionTable(
+        organization=[module_org.id], location=[module_location]
+    ).create()
+    operatingsystem = target_sat.api.OperatingSystem(
+        architecture=[architecture], ptable=[ptable]
+    ).create()
+    medium = target_sat.api.Media(
+        organization=[module_org.id], location=[module_location], operatingsystem=[operatingsystem]
+    ).create()
+    lce = (
+        target_sat.api.LifecycleEnvironment(name="Library", organization=module_org.id)
+        .search()[0]
+        .read()
+        .id
+    )
+    cv = target_sat.api.ContentView(organization=module_org).create()
+    cv.publish()
+
+    # create hostgroup
+    hostgroup_name = gen_string('alpha')
+    target_sat.api.HostGroup(
+        name=hostgroup_name,
+        architecture=architecture,
+        domain=domain,
+        subnet=subnet,
+        location=[module_location.id],
+        medium=medium,
+        operatingsystem=operatingsystem,
+        organization=[module_org],
+        ptable=ptable,
+        lifecycle_environment=lce,
+        content_view=cv,
+        content_source=1,
+    ).create()
+    cr_name = gen_string('alpha')
+    with session:
+        session.computeresource.create(
+            {
+                'name': cr_name,
+                'provider': FOREMAN_PROVIDERS['vmware'],
+                'provider_content.vcenter': module_vmware_settings['vcenter'],
+                'provider_content.user': module_vmware_settings['user'],
+                'provider_content.password': module_vmware_settings['password'],
+                'provider_content.datacenter.value': module_vmware_settings['datacenter'],
+                'locations.resources.assigned': [module_location.name],
+                'organizations.resources.assigned': [module_org.name],
+            }
+        )
+        session.hostgroup.update(hostgroup_name, {'host_group.deploy': cr_name + " (VMware)"})
+        session.computeresource.vm_import(
+            cr_name,
+            module_vmware_settings['vm_name'],
+            hostgroup_name,
+            module_location.name,
+            module_org.name,
+            module_vmware_settings['vm_name'],
+        )
+        host_name = module_vmware_settings['vm_name'] + '.' + domain.name
+        virt_card = session.host_new.get_details(host_name, widget_names='details.virtualization')[
+            'details'
+        ]['virtualization']
+        assert virt_card['datacenter'] == 'RH_Engineering'
+        assert virt_card['cluster'] == 'Satellite-Engineering'
+        assert virt_card['memory'] == '2 GB'
+        assert virt_card['public_ip_address'] == ''
+        assert virt_card['mac_address'] == '00:50:56:8d:a2:96'
+        assert virt_card['cpus'] == '1'
+        assert virt_card['cores_per_socket'] == '1'
+        assert virt_card['firmware'] == 'bios'
+        assert virt_card['hypervisor'] == 'esxi3.satellite.lab.eng.rdu2.redhat.com'
```

Comment:
Of this information, only Datacenter is stored anywhere I have access to. To be able to automate this, we'd need to fetch information seperately from the vmware datacenter, and store that. I suppose I could put this information inside vmware.yml, or a similar test file, but it would be just as fragile as storing it here.

Reply:
One question and one change request: 
- Are the mac_address and the hypervisor deterministically assigned? I.e. if (as I'm assuming) there is more than one ESXi host in the VMware cluster, how do we know it will always be deployed on the specified hypervisor?
- We need to avoid storing RH-internal info in public Git repos. That includes the `datacenter`, `cluster`, `mac_address`, and `hypervisor` values from the test as currently written. Please add those values to the CI config (`vmware.yaml` sounds reasonable to me) and modify this test to check against those settings.